src/satosa/satosa_config.py

```diff
@@ -44,26 +46,49 @@ def __init__(self, config):
 
         self._verify_dict(self._config)
 
+        self._load_plugins()
+        self._load_internal_attributes()
+
+    def _parse_config(self, config):
+        return next(
+            filter(
+                lambda conf: conf is not None,
+                map(
+                    lambda parser: parser(config),
+                    (self._load_dict, self._load_yaml),
+                ),
+            ),
+            None,
+        )
+
+    def _load_plugins(self):
+        def load_plugin_config(config):
+            plugin_config = self._parse_config(config)
+            if not plugin_config:
+                raise SATOSAConfigurationError(
+                    "Failed to load plugin config '{}'".format(config)
+                )
+            else:
+                return plugin_config
+
         # Read plugin configs from dict or file path
         for key in ["BACKEND_MODULES", "FRONTEND_MODULES", "MICRO_SERVICES"]:
-            plugin_configs = []
-            for config in self._config.get(key, []):
-                for parser in parsers:
-                    plugin_config = parser(config)
-                    if plugin_config:
-                        plugin_configs.append(plugin_config)
-                        break
-                else:
-                    raise SATOSAConfigurationError('Failed to load plugin config \'{}\''.format(config))
-            self._config[key] = plugin_configs
-
-        for parser in parsers:
-            _internal_attributes = parser(self._config["INTERNAL_ATTRIBUTES"])
-            if _internal_attributes is not None:
-                self._config["INTERNAL_ATTRIBUTES"] = _internal_attributes
-                break
+            self._config[key] = list(
+                map(
+                    lambda x: load_plugin_config(x),
+                    self._config.get(key, []) or [],
+                )
+            )
+
+    def _load_internal_attributes(self):
+        self._config["INTERNAL_ATTRIBUTES"] = self._parse_config(
+            self._config["INTERNAL_ATTRIBUTES"]
+        )
+
         if not self._config["INTERNAL_ATTRIBUTES"]:
-            raise SATOSAConfigurationError("Could not load attribute mapping from 'INTERNAL_ATTRIBUTES.")
+            raise SATOSAConfigurationError(
+                "Could not load attribute mapping from 'INTERNAL_ATTRIBUTES."
```

Comment:
A third different message for the same action. Can we make those messages similar?